aviatrix/resource_aviatrix_cloudn_edge_gateway.go

```diff
@@ -0,0 +1,300 @@
+				Optional:    true,
+				ForceNew:    true,
+				Description: "DNS server.",
+			},
+			"secondary_dns_server": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Secondary DNS server.",
+			},
+			"image_download_path": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old != ""
+				},
+				Description: "The location where the Edge gateway image will be stored.",
+			},
+			"local_as_number": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				Description:  "Changes the Aviatrix CloudN ASN number before you setup Aviatrix Transit Gateway connection configurations.",
+				ValidateFunc: goaviatrix.ValidateASN,
+			},
+			"prepend_as_path": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "AS path prepend.",
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: goaviatrix.ValidateASN,
+				},
+				MaxItems: 25,
+			},
+		},
+	}
+}
+
+func marshalCloudnEdgeGatewayInput(d *schema.ResourceData) *goaviatrix.CloudnEdgeGateway {
+	cloudnEdgeGateway := &goaviatrix.CloudnEdgeGateway{
+		GatewayName:              d.Get("gw_name").(string),
+		ManagementConnectionType: d.Get("management_connection_type").(string),
+		OverPrivateNetwork:       d.Get("over_private_network").(bool),
+		WanInterfaceIp:           d.Get("wan_interface_ip").(string),
+		WanDefaultGateway:        d.Get("wan_default_gateway").(string),
+		LanInterfaceIp:           d.Get("lan_interface_ip").(string),
+		ManagementInterfaceIp:    d.Get("management_interface_ip").(string),
+		DefaultGatewayIp:         d.Get("default_gateway_ip").(string),
+		DnsServer:                d.Get("dns_server").(string),
+		SecondaryDnsServer:       d.Get("secondary_dns_server").(string),
+		ImageDownloadPath:        d.Get("image_download_path").(string),
+	}
+
+	return cloudnEdgeGateway
+}
+
+func resourceAviatrixCloudnEdgeGatewayCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	cloudnEdgeGateway := marshalCloudnEdgeGatewayInput(d)
+
+	if cloudnEdgeGateway.ManagementConnectionType == "DHCP" && (cloudnEdgeGateway.ManagementInterfaceIp != "" || cloudnEdgeGateway.DefaultGatewayIp != "" ||
+		cloudnEdgeGateway.DnsServer != "" || cloudnEdgeGateway.SecondaryDnsServer != "") {
+		return diag.Errorf("'management_interface_ip', 'default_gateway_ip', 'dns_server' and 'secondary_dns_server' are only valid when 'management_connection_type' is Static")
+	}
+
+	d.SetId(cloudnEdgeGateway.GatewayName)
+	flag := false
+	defer resourceAviatrixCloudnEdgeGatewayReadIfRequired(ctx, d, meta, &flag)
+
+	if err := client.CreateCloundEdgeGateway(ctx, cloudnEdgeGateway); err != nil {
+		return diag.Errorf("could not create cloudn edge gateway: %v", err)
```

Comment:
Should be consistent on how the error messages capitalize the resource. Below you use `CloudN Edge Gateway`.